editor/export/editor_export_plugin.h

```diff
@@ -142,10 +155,20 @@ class EditorExportPlugin : public RefCounted {
 	virtual PackedStringArray _get_export_features(const Ref<EditorExportPlatform> &p_export_platform, bool p_debug) const;
 	virtual void _get_export_options(const Ref<EditorExportPlatform> &p_export_platform, List<EditorExportPlatform::ExportOption> *r_options) const;
 	virtual bool _should_update_export_options(const Ref<EditorExportPlatform> &p_export_platform) const;
+	virtual String _get_export_option_warning(const Ref<EditorExportPlatform> &p_export_platform, const String &p_option_name) const;
 
+public:
 	virtual String _get_name() const;
 
-public:
+	virtual bool _supports_platform(const Ref<EditorExportPlatform> &p_export_platform) const;
+
+	virtual PackedStringArray _get_android_dependencies(const Ref<EditorExportPlatform> &p_export_platform, bool p_debug) const;
+	virtual PackedStringArray _get_android_dependencies_maven_repos(const Ref<EditorExportPlatform> &p_export_platform, bool p_debug) const;
+	virtual PackedStringArray _get_android_libraries(const Ref<EditorExportPlatform> &p_export_platform, bool p_debug) const;
+	virtual String _get_android_manifest_activity_element_contents(const Ref<EditorExportPlatform> &p_export_platform, bool p_debug) const;
+	virtual String _get_android_manifest_application_element_contents(const Ref<EditorExportPlatform> &p_export_platform, bool p_debug) const;
+	virtual String _get_android_manifest_element_contents(const Ref<EditorExportPlatform> &p_export_platform, bool p_debug) const;
+
```

Comment:
These shouldn't be public themselves. If we need to call them from outside of the class, we need public counterparts. Yeah, it's a bit of boilerplate but it's for the sake of code clarity and maintainability. If this is only needed for some small hackish logic within editor/export plugin classes, you can also use class `friend`ness, while keeping the methods private.

Reply:
Or rather, I think C++ virtual methods themselves can be public if that's how they are supposed to be used. Whichever you prefer.